### PR TITLE
mockgcp support for Folders

### PR DESCRIPTION
**Pull request by justinsb**

### CODE REVIEW

**Comment by justinsb** [comment]
/assign @yuwenma 
/milestone 1.125

(I am trying to get our e2e-service-logging tests to pass, amongst others!)

File: pkg/test/resourcefixture/testdata/basic/resourcemanager/v1beta1/folder/folderinfolder/_http.log [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+X-Content-Type-Options: nosniff
+X-Frame-Options: SAMEORIGIN
+X-Xss-Protection: 0
+
+{
+  "name": "operations/${operationID}"
+}
+
+---
+
```
Comment (yuwenma) [inline pkg/test/resourcefixture/testdata/basic/resourcemanager/v1beta1/folder/folderinfolder/_http.log:44]: It looks like TF attempts to get the folder from both v1 and v3 GCP servers. I'm curious why..

**Review by yuwenma — COMMENTED** [review]
/lgtm
/approve

File: pkg/test/resourcefixture/testdata/basic/resourcemanager/v1beta1/folder/folderinfolder/_http.log (hunk 1 above)
Comment (justinsb) [reply pkg/test/resourcefixture/testdata/basic/resourcemanager/v1beta1/folder/folderinfolder/_http.log:44]: Legacy

**Pull request merged**